scripts/flutter_symbol_collector/lib/src/symbol_collector_cli.dart

```diff
@@ -0,0 +1,147 @@
+
+    final archive = ZipDecoder().decodeBytes(zipData);
+    final stream = OutputStream();
+    archive.single.writeContent(stream, freeMemory: true);
+    stream.flush();
+
+    await tempDir.create();
+    final executableFile = await tempDir.childFile(executableName).create();
+    self.cli = executableFile.path;
+
+    await executableFile.writeAsBytes(stream.getBytes(), flush: true);
+    self._log.fine(
+        'Symbol-collector CLI extracted to ${executableFile.path}: ${await executableFile.length()} bytes');
+    self._isExecutable = platform.isWindows;
+    return self;
+  }
+
+  void _ensureIsExecutable() {
+    if (!_isExecutable) {
+      _isExecutable = true;
+      if (LocalPlatform().operatingSystem == platform.operatingSystem) {
+        if (platform.isLinux || platform.isMacOS) {
+          _log.fine('Making Symbol-collector CLI executable (chmod +x)');
+
+          posix.chmod(cli, '0700');
+        }
+      } else {
+        _log.warning(
+            'Symbol-collector CLI has been run with a platform that is not the current OS platform.'
+            'This should only be done in tests because we can\'t execute the downloaded program');
+      }
+    }
+  }
+
+  Future<String> getVersion() => _execute(['--version', '-h']);
+
+  Future<bool> upload(
+      Directory dir, Platform symbolsPlatform, FlutterVersion flutterVersion,
+      {bool dryRun = false}) async {
+    final type = symbolsPlatform.operatingSystem;
+    try {
+      await _execute([
+        '--upload',
+        'directory',
+        '--path',
+        dir.path,
+        '--batch-type',
+        type,
+        '--bundle-id',
+        'flutter-${flutterVersion.tagName}-$type',
+        '--server-endpoint',
+        'https://symbol-collector.services.sentry.io/',
+      ]);
+    } catch (e) {
+      _log.warning('Failed to upload symbols from ${dir.path}', e);
+      return false;
+    }
+    return true;
+  }
+
+  Future<String> _execute(List<String> arguments) async {
+    _ensureIsExecutable();
+
+    _log.fine('Executing ${path.basename(cli)} ${arguments.join(' ')}');
+    final process = await Process.start(cli, arguments);
+
+    final output = StringBuffer();
+    handleOutput(Level level, String message) {
+      message.trimRight().split('\n').forEach((s) => _log.log(level, '   $s'));
+      output.write(message);
+    }
+
+    final pipes = [
+      process.stdout
+          .transform(utf8.decoder)
+          .forEach((s) => handleOutput(Level.FINER, s)),
+      process.stderr
+          .transform(utf8.decoder)
+          .forEach((s) => handleOutput(Level.SEVERE, s))
+    ];
+
+    final exitCode = await process.exitCode;
+    await Future.wait(pipes);
+    final strOutput = output.toString().trimRight();
+    if (exitCode != 0) {
+      throw Exception('Symbol-collector CLI failed with exit code $exitCode.');
+    } else if (strOutput.contains('Exception:')) {
+      // see https://github.com/getsentry/symbol-collector/issues/162
```

Comment:
It's been merged and fixed on 1.6.1: https://github.com/getsentry/symbol-collector/issues/162

Reply:
Doesn't seem to work for all exceptions so I'd keep this:

https://github.com/getsentry/sentry-dart/actions/runs/6649991569/job/18069373897#step:6:4427

```
FINER: 2023-10-26 06:11:27.528091:    info: System.Net.Http.HttpClient.ISymbolClient.LogicalHandler[100]
FINER: 2023-10-26 06:11:27.528112:          Start processing HTTP request POST https://symbol-collector.services.sentry.io/symbol/batch/acb0d8eb-c6f6-46a5-b1c8-176ecb051369/upload
FINER: 2023-10-26 06:11:27.528853:    info: System.Net.Http.HttpClient.ISymbolClient.ClientHandler[100]
FINER: 2023-10-26 06:11:27.528889:          Sending HTTP request POST https://symbol-collector.services.sentry.io/symbol/batch/acb0d8eb-c6f6-46a5-b1c8-176ecb051369/upload
FINER: 2023-10-26 06:11:36.642136:    info: System.Net.Http.HttpClient.ISymbolClient.ClientHandler[101]
FINER: 2023-10-26 06:11:36.642180:          Received HTTP response headers after 9113.6459ms - 400
FINER: 2023-10-26 06:11:36.642633:    info: System.Net.Http.HttpClient.ISymbolClient.LogicalHandler[101]
FINER: 2023-10-26 06:11:36.642644:          End processing HTTP request after 9114.1039ms - 400
FINER: 2023-10-26 06:11:36.654732:    warn: SymbolCollector.Core.SymbolClient[0]
FINER: 2023-10-26 06:11:36.654765:          Failed to upload file
FINER: 2023-10-26 06:11:36.654773:          System.InvalidOperationException: Failed uploading file. Server response: BadRequest
FINER: 2023-10-26 06:11:36.654780:             at SymbolCollector.Core.SymbolClient.ThrowOnUnsuccessfulResponse(String message, HttpResponseMessage checkResponse)
FINER: 2023-10-26 06:11:36.654788:             at SymbolCollector.Core.SymbolClient.<>c__DisplayClass9_0.<<Upload>g__Upload|1>d.MoveNext()
FINER: 2023-10-26 06:11:36.723514:    warn: SymbolCollector.Core.Client[0]
FINER: 2023-10-26 06:11:36.723582:          Failed to upload. Failure count: 1.
FINER: 2023-10-26 06:11:36.723595:          System.InvalidOperationException: Failed uploading file. Server response: BadRequest
FINER: 2023-10-26 06:11:36.723610:             at SymbolCollector.Core.SymbolClient.ThrowOnUnsuccessfulResponse(String message, HttpResponseMessage checkResponse)
FINER: 2023-10-26 06:11:36.723625:             at SymbolCollector.Core.SymbolClient.<>c__DisplayClass9_0.<<Upload>g__Upload|1>d.MoveNext()
FINER: 2023-10-26 06:11:36.723636:          --- End of stack trace from previous location ---
FINER: 2023-10-26 06:11:36.723646:             at SymbolCollector.Core.SymbolClient.<>c__DisplayClass9_0.<<Upload>g__Upload|1>d.MoveNext()
FINER: 2023-10-26 06:11:36.723660:          --- End of stack trace from previous location ---
FINER: 2023-10-26 06:11:36.723670:             at SymbolCollector.Core.SymbolClient.Upload(Guid batchId, String unifiedId, String hash, String fileName, Func`1 fileFactory, CancellationToken token)
FINER: 2023-10-26 06:11:36.723681:             at SymbolCollector.Core.Client.UploadAsync(Guid batchId, ObjectFileResult objectFileResult, CancellationToken cancellationToken)
FINER: 2023-10-26 06:11:36.723692:             at SymbolCollector.Core.Client.UploadFilesAsync(Guid batchId, String path, CancellationToken cancellationToken)
```

Reply:
Those `warn` should be retries, and at the end (3?) it should error

Reply:
good point, done